### PR TITLE
Removed SIG form

### LINKED ISSUE #116: Get involved link should point to the mailing list
This section:

> Get Involved
> There is a lot to do! If you are interested in getting involved, please get in touch via this simple form. We are a friendly, collaborative group and look forward to working together.

should point to the mailing list, not some custom form. See https://github.com/open-telemetry/community/pull/108

### PR DESCRIPTION
**Pull request by mtwo**
Fixes #116

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ OpenTelemetry community content.
 
 ## Get Involved
 
-There is a lot to do! If you are interested in getting involved, please join the mailing lists and attend the community meetings. If you wish to join one of the special interest groups, you can get in touch via [this simple form](https://forms.gle/HCGGoLRzoPm6o29K6). We are a friendly, collaborative group and look forward to working together.
```
Comment (bhs) [inline README.md]: Minor point, but I kinda liked the "we are a friendly <blah blah blah>" part of the previous message, or at least the spirit of it. Generally I just want to make it psychologically easy for people without a social connection to existing OpenTel community members to get involved.

**Comment by mtwo** [comment]
Sounds good! I’ll update

________________________________

File: README.md [hunk 2]
```diff
@@ -4,7 +4,7 @@ OpenTelemetry community content.
...
+There is a lot to do! If you are interested in getting involved, please join the mailing lists and attend the community meetings. If you're interested in contributing to a specific part of the project, please join the appropriate special interest group (SIG). Details for all of these items are below.
```
Comment (mtwo) [inline README.md]:
```suggestion
There is a lot to do! If you are interested in getting involved, please join the mailing lists and attend the community meetings. If you're interested in contributing to a specific part of the project, please join the appropriate special interest group (SIG). Details for all of these items are below. We are a friendly, collaborative group and look forward to working together!
```

**Comment by mtwo** [comment]
@bhs added!

**Comment by SergeyKanzhelev** [comment]
@bhs please comment if you think more changes are needed

**Comment by bhs** [comment]
LGTM, FWIW. Thanks, @mtwo!

**Pull request merged**